src/validators/domain.py

```diff
@@ -59,17 +59,21 @@ def domain(
         return False
 
     try:
-        return not re.search(r"\s", value) and re.match(
+
+        service_record = r"_" if rfc_2782 else ""
+        trailing_dot = r"\.?$" if rfc_1034 else r"$"
+
+        return not re.search(r"\s|__+", value) and re.match(
```

Comment:
I'm not as familiar with rfc 2782, is there something in it forbidding consecutive underscores?

Reply:
As I understand, [RFC 2782](https://www.rfc-editor.org/rfc/rfc2782) specifies underscores for service record type of domains, which is a very special cases considering domain names do not generally contain underscores (even though the aren't prohibited).

Allowing consecutive underscores would allow weird domain names like: `___exa__mple___`. On another note, browsers seem to work file with such domain names in the URL.